### PR TITLE
Added framebuffer and related attributes to XRWebGLLayer

### PR DESCRIPTION
**Pull request by asajeffrey**
Implement the `framebuffer` attribute of `XRWebGLLayer`.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes have tests

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23797)

### CODE REVIEW

**Comment by asajeffrey** [comment]
The bit i'm not sure about is
```rust
        // Step 8.3. "Allocate and initialize resources compatible with session’s XR device,
        // including GPU accessible memory buffers, as required to support the compositing of layer."
        let texture = context.CreateTexture().ok_or(Error::Operation)?;
        let resolution =
            session.with_session(|session| session.recommended_framebuffer_resolution());
        let mut pixels = CustomAutoRooter::new(None);
        context.TexImage2D(
            constants::TEXTURE_2D,
            0,
            constants::RGBA,
            resolution.width,
            resolution.height,
            0,
            constants::RGBA,
            constants::UNSIGNED_BYTE,
            pixels.root(cx),
        )?;
        context.FramebufferTexture2D(
            constants::FRAMEBUFFER,
            constants::COLOR_ATTACHMENT0,
            constants::TEXTURE_2D,
            Some(&texture),
            0,
        );
```
which is a bit random. We can't make this up to the device, slightly annoyingly, as it needs to be run in WebGL, not GL, otherwise the GL state and the WebGL state will get out of sync.

**Comment by asajeffrey** [comment]
The matching webxr PR is https://github.com/servo/webxr/pull/19

**Comment by asajeffrey** [comment]
Added tests. @Manishearth this is ready for review!

File: components/script/dom/webidls/DOMException.webidl [hunk 1]
```diff
@@ -38,7 +38,10 @@ interface DOMException {
   const unsigned short TIMEOUT_ERR = 23;
   const unsigned short INVALID_NODE_TYPE_ERR = 24;
   const unsigned short DATA_CLONE_ERR = 25;
+  // Only the first 25 errors are given codes in
+  // https://heycam.github.io/webidl/#idl-DOMException
   const unsigned short NOT_READABLE_ERR = 26;
+  const unsigned short OPERATION_ERR = 27;
```
Comment (Manishearth) [inline components/script/dom/webidls/DOMException.webidl]: should we be exposing this? I don't see these in Firefox.

**Review by Manishearth — APPROVED** [review]

**Comment by asajeffrey** [comment]
cc @alcooper91 re the webxr WPT tests. I think the changes in https://github.com/servo/servo/pull/23797/commits/d6d041671321a134a8693e39428821008407c413 are spec-compliant. The only bit I wasn't sure about is the expected semantics of `resolution` in `FakeXRViewInit`, which I took to mean the resolution of each eye (i.,e. should be the size of the eye's viewport) not the native resolution of the whole device (https://immersive-web.github.io/webxr/#native-webgl-framebuffer-resolution).

**Comment by asajeffrey** [comment]
@jdm approved the changes to DOMException on IRC: https://mozilla.logbot.info/servo/20190718#c16478495

@bors-servo: r=Manishearth

**Comment by jdm** [comment]
@bors-servo retry
* #23290

**Pull request merged**